### Step 1. 필요한 라이브러리 임포트

In [0]:
import pandas as pd

### Step 2. [데이터](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user) 확인

- user_id 유저 아이디
- age 나이
- gender 성별
- occupation 직업
- zip_code 우편번호

### Step 3. 

**read_csv를 통해 데이터를 올바른 형태로 불러온 뒤 users에 저장해주세요.**

user_id를 인덱스로 설정해주세요.

In [0]:
url = 'https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user'

users = pd.read_csv(url, sep='|', index_col='user_id')

### Step 4. 초반 25행 조회

In [6]:
users.head(25)

age gender     occupation zip_code
user_id                                    
1         24      M     technician    85711
2         53      F          other    94043
3         23      M         writer    32067
4         24      M     technician    43537
5         33      F          other    15213
6         42      M      executive    98101
7         57      M  administrator    91344
8         36      M  administrator    05201
9         29      M        student    01002
10        53      M         lawyer    90703
11        39      F          other    30329
12        28      F          other    06405
13        47      M       educator    29206
14        45      M      scientist    55106
15        49      F       educator    97301
16        21      M  entertainment    10309
17        30      M     programmer    06355
18        35      F          other    37212
19        40      M      librarian    02138
20        42      F      homemaker    95660
21        26      M         writer    30068
22        25      M         writer    40206
23        30      F         artist    48197
24        21      F         artist    94533
25        39      M       engineer    55107

### Step 5. 마지막 10행 조회

In [7]:
users.tail(10)

age gender     occupation zip_code
user_id                                    
934       61      M       engineer    22902
935       42      M         doctor    66221
936       24      M          other    32789
937       48      M       educator    98072
938       38      F     technician    55038
939       26      F        student    33319
940       32      M  administrator    02215
941       20      M        student    97229
942       48      F      librarian    78209
943       22      M        student    77841

### Step 6. 데이터셋은 몇 개의 행으로 이루어져 있나요?

943

In [8]:
users.shape[0]

943

### Step 7.  데이터셋은 몇 개의 열로 이루어져 있나요?

4

In [9]:
users.shape[1]

4

### Step 8. 열(column)의 이름들을 모두 조회해주세요

Index(['age', 'gender', 'occupation', 'zip_code'], dtype='object')

In [10]:
users.columns

Index(['age', 'gender', 'occupation', 'zip_code'], dtype='object')

### Step 9. 데이터셋의 인덱스를 조회해주세요

Int64Index([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,
            ...
            934, 935, 936, 937, 938, 939, 940, 941, 942, 943],
           dtype='int64', name='user_id', length=943)

In [11]:
# "the index" (aka "the labels")
users.index

Int64Index([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,
            ...
            934, 935, 936, 937, 938, 939, 940, 941, 942, 943],
           dtype='int64', name='user_id', length=943)

### Step 10. 각 열의 데이터타입은 무엇인가요?

In [12]:
users.dtypes

age            int64
gender        object
occupation    object
zip_code      object
dtype: object

### Step 11. occupation column만 조회해주세요

In [13]:
users.occupation

#or

users['occupation']

user_id
1         technician
2              other
3             writer
4         technician
5              other
           ...      
939          student
940    administrator
941          student
942        librarian
943          student
Name: occupation, Length: 943, dtype: object

### Step 12. 데이터셋에는 몇 종류의 직업이 존재하나요?

21

In [14]:
users.occupation.nunique()
#or by using value_counts() which returns the count of unique elements
#users.occupation.value_counts().count()

21

### Step 13. 데이터셋에서 가장 많이 존재하는 직업은 무엇인가요?

student

In [15]:
#Because "most" is asked
users.occupation.value_counts().head(1).index[0]

#or
#to have the top 5

# users.occupation.value_counts().head()

'student'

### Step 14. users.describe() 코드를 실행해서 숫자 열의 요약정보를 조회해주세요

In [16]:
users.describe() #Notice: by default, only the numeric columns are returned. 

age
count  943.000000
mean    34.051962
std     12.192740
min      7.000000
25%     25.000000
50%     31.000000
75%     43.000000
max     73.000000

### Step 15. users.describe(include = 'all') 코드를 실행해서 모든 열의 정보를 조회해주세요

In [17]:
users.describe(include = "all") #Notice: By default, only the numeric columns are returned.

age gender occupation zip_code
count   943.000000    943        943      943
unique         NaN      2         21      795
top            NaN      M    student    55414
freq           NaN    670        196        9
mean     34.051962    NaN        NaN      NaN
std      12.192740    NaN        NaN      NaN
min       7.000000    NaN        NaN      NaN
25%      25.000000    NaN        NaN      NaN
50%      31.000000    NaN        NaN      NaN
75%      43.000000    NaN        NaN      NaN
max      73.000000    NaN        NaN      NaN

### Step 16. occupation 칼럼만 조회해서 describe() 함수를 실행해주세요

count         943<br>
unique         21<br>
top       student<br>
freq          196<br>
Name: occupation, dtype: object<br>

In [18]:
users.occupation.describe()

count         943
unique         21
top       student
freq          196
Name: occupation, dtype: object

### Step 17. user들의 평균 나이는 몇 살인가요?
34

In [19]:
round(users.age.mean())

34

### Step 18. 데이터셋에서 가장 적게 등장하는 나이는 몇 살인가요?

가장 적게 등장하는 나이는 1번씩 등장하는 나이이며  7, 66, 73, 10, 11살 입니다. 이러한 결과를 조회할수 있도록 데이터 조회하는 코드를 작성해주세요.

In [20]:
users.age.value_counts().tail() #7, 10, 11, 66 and 73 years -> only 1 occurrence

11    1
10    1
73    1
66    1
7     1
Name: age, dtype: int64